release.sh

```diff
@@ -0,0 +1,405 @@
+#! /bin/sh
+
+# Author:  Thomas DEBESSE <dev@illwieckz.net>
+# License: MIT
+
+# be safe
+sync
+
+game_name='Unvanquished'
+
+engine_file_list='irt_core-x86_64.nexe'
+engine_bin_list='daemon daemonded daemon-tty'
+engine_extra_bin_list='crash_server nacl_loader nacl_helper_bootstrap'
+engine_archive_fmt='zip'
+
+vm_kind_list='cgame sgame'
+vm_arch_list='x86 x86_64'
+
+root_dir="$(git rev-parse --show-toplevel)"
+build_dir="${root_dir}/build"
+release_dir="${build_dir}/release"
+
+symbol_basename='symbols'
+symbol_dir="${build_dir}/${symbol_basename}"
+
+symbol_archive_fmt='zip'
+symbol_archive="${build_dir}/${symbol_basename}.${symbol_archive_fmt}"
+
+vmpak_archive_fmt='zip'
+vmpak_extension='pk3'
+vmpak_basename='vm'
+vmpak_dir="${build_dir}/vm"
+
+engine_subdir='engine'
+engine_dir="${root_dir}/build/${engine_subdir}"
+
+main_nexe='main.nexe'
+
+build_vm='false'
+build_linux64='false'
+build_linux='false'
+build_mac64='false'
+build_mac='false'
+build_win64='false'
+build_win='false'
+build_breakpad='false'
+engine_target=''
+
+_error () {
+	printf 'ERROR: %s\n' "${1}" >&2
+	exit 1
+}
+
+_help () {
+	tab="$(printf '\t')"
+	cat <<-EOF
+	${0}: a tool to build release files for ${game_name}.
+
+	Usage:
+
+	${tab}${0} <target>
+
+	The script *must* be run from ${game_name} source directory,
+	but can be called from everywhere in source directory.
+
+	Target can be:
+
+	${tab}vm
+	${tab}${tab}build virtual machine code
+
+	${tab}linux32
+	${tab}${tab}build linux32 engine binaries (no cross compilation)
+
+	${tab}linux64
+	${tab}${tab}build linux64 engine binaries
+
+	${tab}mac32
+	${tab}${tab}build mac32 engine binaries (untested, likely uncomplete)
+
+	${tab}mac64
+	${tab}${tab}build mac64 engine binaries (untested, likely uncomplete)
+
+	${tab}win32
+	${tab}${tab}build win32 engine binaries (untested, likely uncomplete)
+
+	${tab}win64
+	${tab}${tab}build win64 engine binaries (untested, likely uncomplete)
+
+	Example:
+
+	${tab}${0} vm linux64
+
+	EOF
+	exit
+}
+
+[ -f "${root_dir}/src/cgame/cg_main.cpp" ] || _error "must be run from ${game_name} source tree"
+
+[ -z "${1}" ] && _help
+
+while ! [ -z "${1}" ]
+do
+	case "${1}" in
```

Comment:
for a proper release, it would be a good idea to use the basepak.sh script instead. That will include the vms too and handle stripping them and all that jazz.

Reply:
the idea there is just to produce a pk3 that can be merged to another pk3.